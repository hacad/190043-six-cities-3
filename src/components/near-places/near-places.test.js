import React from "react";
import renderer from "react-test-renderer";
import NearPlaces from "./near-places.jsx";

jest.mock(`../place-card/place-card.jsx`, () => `<div />`);

it(`NearPlaces correctly renders after relaunch`, () => {
  const places = [{
    id: 15,
    city: {
      name: `Hamburg`,
      location: {
        latitude: 53.552645,
        longitude: 9.966287,
        zoom: 10
      }
    },
    type: `apartment`,
    previewImage: `img/apartment-03.jpg`,
    isPremium: false,
    title: `Nice, cozy, warm big bed apartment`,
    price: 180,
    rating: 100,
    isFavorite: false,
    location: {
      latitude: 53.554221,
      longitude: 9.992588
    }
  }];

  const activeItem = {
    id: 16,
    city: {
      name: `Paris`,
      location: {
        latitude: 48.856663,
        longitude: 2.351556,
        zoom: 10
      }
    },
    type: `apartment`,
    previewImage: `img/apartment-02.jpg`,
    isPremium: false,
    title: `Nice, cozy, warm big bed apartment`,
    price: 300,
    rating: 100,
    isFavorite: false,
    location: {
      latitude: 48.845933,
      longitude: 2.357839
    }
  };

  const nearPlaces = renderer.create(
      <NearPlaces
        places={places}
        activeCity={activeItem.city}
        currentOfferLocation={activeItem.location}
        activeItem={activeItem}
        onActivateItem={jest.fn()}
        onDeactivateItem={jest.fn()}
      />
  ).toJSON();

  // Act & Assert
  expect(nearPlaces).toMatchSnapshot();
});
