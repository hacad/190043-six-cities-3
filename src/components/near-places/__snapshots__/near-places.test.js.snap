// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NearPlaces correctly renders after relaunch 1`] = `
Array [
  <section
    className="property__map map"
    id="map"
  />,
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <div
        className="near-places__list places__list"
      >
        <<div />
          onActivate={[MockFunction]}
          onClickHeader={[Function]}
          onDeactivate={[MockFunction]}
          place={
            Object {
              "city": Object {
                "location": Object {
                  "latitude": 53.552645,
                  "longitude": 9.966287,
                  "zoom": 10,
                },
                "name": "Hamburg",
              },
              "id": 15,
              "isFavorite": false,
              "isPremium": false,
              "location": Object {
                "latitude": 53.554221,
                "longitude": 9.992588,
              },
              "previewImage": "img/apartment-03.jpg",
              "price": 180,
              "rating": 100,
              "title": "Nice, cozy, warm big bed apartment",
              "type": "apartment",
            }
          }
        />
      </div>
    </section>
  </div>,
]
`;
